src/formatDate.js

```diff
@@ -7,8 +7,31 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(dateString, fromFormat, toFormat) {
+  const dateArray = dateString.split(fromFormat[3]);
+  const object = {};
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    object[fromFormat[i]] = dateArray[i];
+  }
+
+  if (object.YYYY) {
+    object.YY = object.YYYY.slice(2);
+  }
+
+  if (object.YY < 30) {
+    object.YYYY = '20' + `${object.YY}`;
+  } else {
+    object.YYYY = '19' + `${object.YY}`;
+  }
+
+  for (let i = 0; i < 3; i++) {
+    newDate.push(object[toFormat[i]]);
+  }
+
+  return newDate.join(toFormat[toFormat.length - 1]);
```

Comment:
The use of `toFormat.length - 1` to access the separator is not safe. If the `toFormat` array does not contain a separator as the last element, this will result in an incorrect format. Instead, explicitly check for the separator in the `toFormat` array.